install/charts/templates/deployment.yaml

```diff
@@ -32,6 +32,18 @@ spec:
         app: {{ .Values.metadata.labelApp }}
     spec:
       serviceAccount: {{ .Values.rbac.serviceAccount }}
+      affinity:
+        podAntiAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
+          - weight: 100
+            podAffinityTerm:
+              labelSelector:
+                matchExpressions:
+                - key: app
+                  operator: In
+                  values:
+                  - {{ .Values.metadata.labelApp }}
+              topologyKey: "node"
```

Comment:
What happens if there is only one node? Will only one replica start and the others remain pending? I think we should achieve the following: If there is a node for each replica then distribute them. If there are less nodes then still start all configured replicas (even if this means that on some nodes there is > 1 replica).

Reply:
In my understanding, you are describing what the `preferredDuringSchedulingIgnoredDuringExecution` should do.

As written in [docs](https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity): "`preferredDuringSchedulingIgnoredDuringExecution` would be try to run this set of pods in failure zone XYZ, but if it’s not possible, then allow some to run elsewhere."

As far as I tested it, the Karydia pods will be spread over multiple nodes when possible. But if i.e. only one node is available, all three replicas will be deployed on the same node.

Reply:
Okay, thanks, great!

Reply:
Okay, thanks, great!